test/jump_diff_param.jl

```diff
@@ -0,0 +1,407 @@
+# using Revise
+
+using JuMP
+using DiffOpt
+using Test
+import ParametricOptInterface as POI
+using HiGHS
+using SCS
+
+function test_diff_rhs()
+    model = Model(() -> POI.Optimizer(DiffOpt.Optimizer(HiGHS.Optimizer())))
+    set_silent(model)
+    @variable(model, x)
+    @variable(model, p in MOI.Parameter(3.0))
+    @constraint(model, cons, x >= 3 * p)
+    @objective(model, Min, 2x)
+    optimize!(model)
+    @test MOI.get(model, MOI.VariablePrimal(), x) ≈ 9
+    # the function is
+    # x(p) = 3p, hence x'(p) = 3
+    # differentiate w.r.t. p
+    MOI.set(model, POI.ForwardParameter(), p, 1)
+    DiffOpt.forward_differentiate!(model)
+    @test MOI.get(model, DiffOpt.ForwardVariablePrimal(), x) ≈ 3
+    # again with different "direction"
+    MOI.set(model, POI.ForwardParameter(), p, 2)
+    DiffOpt.forward_differentiate!(model)
+    @test MOI.get(model, DiffOpt.ForwardVariablePrimal(), x) ≈ 6
+    #
+    MOI.set(model, POI.ParameterValue(), p, 2.0)
+    optimize!(model)
+    @test MOI.get(model, MOI.VariablePrimal(), x) ≈ 6
+    # differentiate w.r.t. p
+    MOI.set(model, POI.ForwardParameter(), p, 1)
+    DiffOpt.forward_differentiate!(model)
+    @test MOI.get(model, DiffOpt.ForwardVariablePrimal(), x) ≈ 3
+    # again with different "direction"
+    MOI.set(model, POI.ForwardParameter(), p, 2)
+    DiffOpt.forward_differentiate!(model)
+    @test MOI.get(model, DiffOpt.ForwardVariablePrimal(), x) ≈ 6
+    #
+    # test reverse mode
+    #
+    MOI.set(model, DiffOpt.ReverseVariablePrimal(), x, 1)
+    DiffOpt.reverse_differentiate!(model)
+    @test MOI.get(model, POI.ReverseParameter(), p) ≈ 3
+    # again with different "direction"
+    MOI.set(model, DiffOpt.ReverseVariablePrimal(), x, 2)
+    DiffOpt.reverse_differentiate!(model)
+    @test MOI.get(model, POI.ReverseParameter(), p) ≈ 6
+    return
+end
+
+function test_diff_vector_rhs()
+    model = direct_model(POI.Optimizer(DiffOpt.diff_optimizer(SCS.Optimizer)))
+    set_silent(model)
+    @variable(model, x)
+    @variable(model, p in MOI.Parameter(3.0))
+    @constraint(model, cons, [x - 3 * p] in MOI.Zeros(1))
+
+    # FIXME
+    @constraint(model, fake_soc, [0, 0, 0] in SecondOrderCone())
+
+    @objective(model, Min, 2x)
+    optimize!(model)
```

Comment:
@blegat , I had to add this fake SOC constraint to force it into the Conic model. Otherwise, it tries to use a quadratic model (as the model is linear) and then fails with an error about not being able to push the DiffOpt attributes through the scalarize bridge.

Reply:
Many bridges are missing in DiffOpt but it's easy to add, open an issue

Reply:
You can use `DiffOpt.ModelConstructor` to force it to use conic